error.go

```diff
@@ -397,3 +397,46 @@ func Append(left error, right error) error {
 	errors := [2]error{left, right}
 	return fromSlice(errors[0:])
 }
+
+// AppendInto appends an error into the destination of an error pointer and
+// returns whether the error being appended was non-nil.
+//
+// 	var err error
+// 	multierr.AppendInto(&err, r.Close())
+// 	multierr.AppendInto(&err, w.Close())
+//
+// The above is equivalent to,
+//
+// 	err := multierr.Append(r.Close(), w.Close())
+//
+// As AppendInto reports whether the provided error was non-nil, it may be
+// used to build a multierr error in a loop more ergonomically.
+//
+// 	var err error
+// 	for line := range lines {
+// 		var item Item
+// 		if multierr.AppendInto(&err, parse(line, &item)) {
+// 			continue
```

Comment:
nit: I'd write this in the negative, ala "if-not-append"

Reply:
I'd like to keep this 1:1 comparable to the non-AppendInto version for which `err != nil` would be more standard.

I can add more to the success case if that would alleviate this concern.